2022-01-16 18:34:09   Arguments: Namespace(upscale_input=False, downscale_input=False, augment_input='None', backbone='resnet50', use_netvlad=True, netvlad_clusters=64, use_gem=None, gem_p=3, gem_eps=1e-06, use_sgd=None, use_adagrad=None, momentum=0.9, train_batch_size=4, infer_batch_size=16, margin=0.1, epochs_num=20, patience=3, lr=1e-05, cache_refresh_rate=1000, queries_per_epoch=5000, negs_num_per_query=10, neg_samples_num=1000, seed=0, device='cuda', num_workers=12, val_positive_dist_threshold=25, train_positives_dist_threshold=10, recall_values=[1, 5, 10, 20], datasets_folder='./Dataset_pitts30k', exp_name='default', test_dataset_name=None, test_model_path=None, output_folder='runs/default/2022-01-16_18-34-09')
2022-01-16 18:34:09   The outputs are being saved in runs/default/2022-01-16_18-34-09
2022-01-16 18:34:09   Using 1 GPUs and 12 CPUs
2022-01-16 18:34:09   Loading dataset Pitts30k from folder ./Dataset_pitts30k
2022-01-16 18:34:09   There are 96 queries without any positives within the training set. They won't be considered as they're useless for training.
2022-01-16 18:34:09   Train query set: < TripletsDataset, pitts30k - #database: 10000; #queries: 7320 >
2022-01-16 18:34:09   Val set: < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >
2022-01-16 18:34:09   Test set: < BaseDataset, pitts30k - #database: 10000; #queries: 6816 >
2022-01-16 18:34:10   Train only conv5 of the ResNet-50, freeze the previous ones
2022-01-16 18:34:21   Output dimension of the model is 131072
2022-01-16 18:34:21   Start training epoch: 00
2022-01-16 18:34:21   Cache: 0 / 5
2022-01-16 18:41:46   
Traceback (most recent call last):
  File "/home/franzhd/Aml_project/project_vg/run_train.py", line 100, in <module>
    features = model(images.to(args.device))
  File "/home/franzhd/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/franzhd/Aml_project/project_vg/network.py", line 46, in forward
    x = self.aggregation(x)
  File "/home/franzhd/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/franzhd/Aml_project/project_vg/netvlad.py", line 64, in forward
    residual = x_flatten.unsqueeze(0).permute(1, 0, 2, 3) - \
RuntimeError: CUDA out of memory. Tried to allocate 114.00 MiB (GPU 0; 7.77 GiB total capacity; 5.06 GiB already allocated; 39.19 MiB free; 5.27 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

